### PR TITLE
Feature/cat 2799 rigorously parse supportedchains response from cas

### PR DESCRIPTION
**Pull request by samika98**
## Description

Implemented a validation mechanism in the response processing from when js-ceramic queries CAS to get supportedChain information . This update handles cases where only one supported chain is expected as response from CAS. 

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.
Testing was done through **unit tests.** The unit tests cover both positive and negative flows.

**1. Positive Tests**

- Single Supported Chain: Ensured the validation passes when the 'supportedChains' field in the CAS response is an array containing exactly one element.

**2. Negative Tests**

- Multiple Chains: Checked if the validation fails when 'supportedChains' contains more than one chain.
- Null or Empty Value: Tested for failure when 'supportedChains' has a null or an empty value.
- Missing 'supportedChains' Field: Validated that the absence of the 'supportedChains' field in the CAS response leads to a failure.

## PR checklist

Before submitting this PR, please make sure:

- [ Yes] I have tagged the relevant reviewers and interested parties
- [Not needed ] I have updated the READMEs of affected packages
- [Not needed ] I have made corresponding changes to the documentation

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.
Used **_Codeco_** for validations. See Library here : https://github.com/ceramicnetwork/codeco

### CODE REVIEW

File: packages/codecs/src/anchor.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export enum AnchorRequestStatusName {
 
 /**
  * Part of CAS response that sends AnchorCommit content. Effectively a historical artefact.
+ * sparse datatype : some of the key values can be missing
```
Comment (samika98) [inline packages/codecs/src/anchor.ts:18]: remove

File: packages/codecs/src/anchor.ts [hunk 2]
```diff
@@ -72,3 +73,16 @@ export type ErrorResponse = TypeOf<typeof ErrorResponse>
 
 export const CASResponseOrError = union([CASResponse, ErrorResponse], 'CASResponseOrError')
 export type CASResponseOrError = TypeOf<typeof CASResponseOrError>
+
+
```
Comment (samika98) [inline packages/codecs/src/anchor.ts:77]:
```
`npm run lint`
`npm run prettier`
```

**Review by stbrody — COMMENTED** [review]
one small nitpick but mostly looks great!

File: packages/core/src/anchor/ethereum/remote-cas.ts [hunk 3]
```diff
@@ -75,7 +80,16 @@ export class RemoteCAS implements CASClient {
...
     const response = await this.#sendRequest(this.#chainIdApiEndpoint)
-    return response.supportedChains as Array<string>
+    try {
+      const supportedChainsResponse = decode(SupportedChainsResponse, response)
+      return supportedChainsResponse.supportedChains
+    } catch (error) {
+      throw new Error(
+        `SupportedChains response : ${JSON.stringify(
+          response
+        )} does not contain contain the field <supportedChains> or is of size more than 1`
```
[the remark this answers is not in the document]
Comment (samika98) [reply packages/core/src/anchor/ethereum/remote-cas.ts]: Added suggestion 👍

**Review by ukstv — APPROVED** [review]
Noice!

**Pull request merged**